EFM/source/RAPTOR.h

```diff
@@ -0,0 +1,294 @@
+        left_engine_integrity + right_engine_integrity) / 5;
+
+    bool invincible = true, infinite_fuel = false, easy_flight = false;
+    double shake_amplitude = 0, fm_clock = 0;
+    bool sim_initialised = false;
+
+    // FBW state variables
+    double ref_roll = 0, ref_heading = 0;
+    double roll_error_i = 0, yaw_error_i = 0;
+    double last_aileron_cmd = 0, last_elevator_cmd = 0, last_rudder_cmd = 0;
+    bool autotrim_active = false;
+    double last_g = 0.0;
+    double autotrim_elevator_cmd = 0.0;
+    double takeoff_trim_cmd = 0.0;
+    double g_limit_positive = 12.0;
+    double g_limit_negative = -4.5;
+    double Kp_g = 0.1;
+    double tv_angle = 0.0;
+
+    // Global variables for state persistence
+    double last_yaw_input = 0.0;
+    double last_tv_cmd = 0.0;
+    double last_pitch_input = 0.0;
+    double ground_speed_knots = 0.0;
+
+    //Engine Start Sequence
+    enum EngineState { OFF, STARTING, RUNNING, SHUTDOWN };
+    EngineState left_engine_state = OFF;
+    EngineState right_engine_state = OFF;
+    double left_engine_start_timer = 0.0;
+    double right_engine_start_timer = 0.0;
+
+    // Engine start parameters
+    const double engine_start_time = 30.0;
+    const double starter_phase_duration = 8.0;
+    const double ignition_phase_duration = 12.0;
+    const double spool_up_phase_duration = 10.0;
+    const double starter_rpm = 0.2;
+    const double ignition_rpm = 0.4;
+    const double starter_rate = starter_rpm / starter_phase_duration;
+    const double ignition_rate = (ignition_rpm - starter_rpm) / ignition_phase_duration;
+    const double spool_up_rate = (idle_rpm - ignition_rpm) / spool_up_phase_duration;
+
+    //Lights
+    bool taxi_lights = false;
+    bool landing_lights = false;
+    bool form_light = false;
+    bool nav_white = false;
+    bool anti_collision = false;
+    bool aar_light = false;
+    bool nav_lights = false;
+    double anti_collision_timer = 0.0;
+    bool anti_collision_blink = false;
+    double nav_white_timer = 0.0;
+    bool nav_white_blink = false;
+    double nav_lights_timer = 0.0;
+    bool nav_lights_blink = false;
+    const double blink_on_time = 1.0;
+    const double blink_off_time = 1.0;
+    const double blink_period = blink_on_time + blink_off_time;
+}
+
+// ----- Cockpit Logic -----
+
+class CockpitManager {
+public:
+    CockpitManager();
+    ~CockpitManager() = default;
+
+    // Initialize parameter handles and states for cold/hot start
+    void initialize();
+
+    // Update logic at 0.006s intervals (ed_fm_simulate rate)
+    void update(double dt);
+
+    // Set cockpit draw arguments (lighting, switches)
+    void updateDrawArgs(float* drawargs, size_t size);
+
+private:
+    EDPARAM param_api_; // Parameter API
+    std::map<std::string, void*> param_handles_; // Handles for APU_POWER, AAR, AAR_READY, etc.
+    std::map<int, float> draw_args_; // Input args (709, 712-715), output args (604-613)
```

Comment:
not super important but unless you need these ordered you can change these to unordered_map (faster)